server/src/infra/repositories/search.repository.ts

```diff
@@ -65,8 +64,11 @@ export class SearchRepository implements ISearchRepository {
     let builder = this.assetRepository.createQueryBuilder('asset');
     builder = searchAssetBuilder(builder, options);
 
-    builder.orderBy('asset.fileCreatedAt', options.orderDirection ?? 'DESC');
+    if (options.userIds) {
+      builder.andWhere('asset.ownerId IN (:...userIds)', { userIds: options.userIds });
+    }
```

Comment:
Both search endpoints use `userIds`. This can probably go in the search builder and replace the `ownerId` field there.